scripts/check-site.sh

```
@@ -0,0 +1,52 @@
+    route_path="${route#/}"
+    route_path="${route_path%/}"
+    page="${SITE_DIR}/${route_path}/index.html"
+  fi
+  [[ -f "$page" ]] || report_failure "sitemap route is not staged: ${route}"
+done < <(sed -n 's#.*<loc>https://mcgeer.dev\(/[^<]*\)</loc>.*#\1#p' "${ROOT_DIR}/sitemap.xml")
+
+while IFS= read -r html; do
+  canonical="$(sed -n 's#.*<link rel="canonical" href="https://mcgeer.dev\([^"]*\)".*#\1#p' "$html" | head -n 1)"
+  [[ -z "$canonical" ]] && continue
+  if ! grep -q "<loc>https://mcgeer.dev${canonical}</loc>" "${ROOT_DIR}/sitemap.xml"; then
+    report_failure "canonical URL is missing from sitemap: ${canonical} (${html#${SITE_DIR}/})"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)